framework/src/modules/interoperability/mainchain/module.ts

```diff
@@ -227,4 +244,191 @@ export class MainchainInteroperabilityModule extends BaseInteroperabilityModule
+		const mainchainID = getMainchainID(ctx.chainID);
+		const chainDataNames = chainInfos.map(info => info.chainData.name);
+
+		// verify root level properties
+		for (const chainInfo of chainInfos) {
+			const { chainID } = chainInfo;
+
+			// chainInfo.chainID != getMainchainID();
+			if (chainID.equals(mainchainID)) {
+				throw new Error(`chainID must be not equal to getMainchainID().`);
+			}
+
+			// - chainInfo.chainId[0] == getMainchainID()[0].
+			if (chainID[0] !== mainchainID[0]) {
+				throw new Error(`chainID[0] doesn't match getMainchainID()[0].`);
+			}
+
+			this._verifyChainData(ctx, chainInfo, chainDataNames);
+			this._verifyChannelData(ctx, chainInfo);
+			this._verifyChainValidators(chainInfo);
+		}
+	}
+
+	private _verifyChainData(
+		ctx: GenesisBlockExecuteContext,
+		chainInfo: ChainInfo,
+		chainDataNames: string[],
+	) {
+		const validStatuses = [ChainStatus.REGISTERED, ChainStatus.ACTIVE, ChainStatus.TERMINATED];
+
+		// The entries chainData.name must be pairwise distinct
+		if (new Set(chainDataNames).size !== chainDataNames.length) {
+			throw new Error(`chainData.name must be pairwise distinct.`);
+		}
+
+		const { chainData } = chainInfo;
+
+		// chainData.lastCertificate.timestamp < g.header.timestamp;
+		if (chainData.lastCertificate.timestamp > ctx.header.timestamp) {
+			throw new Error(`chainData.lastCertificate.timestamp must be less than header.timestamp.`);
+		}
+
+		// chainData.name only uses the character set a-z0-9!@$&_.;
+		if (!isValidName(chainData.name)) {
+			throw new Error(`chainData.name only uses the character set ${validNameCharset}.`);
+		}
+
+		// chainData.status is in set {CHAIN_STATUS_REGISTERED, CHAIN_STATUS_ACTIVE, CHAIN_STATUS_TERMINATED}.
+		if (!validStatuses.includes(chainData.status)) {
+			throw new Error(`chainData.status must be one of ${validStatuses.join(', ')}`);
+		}
+	}
+
+	private _verifyChannelData(ctx: GenesisBlockExecuteContext, chainInfo: ChainInfo) {
+		const mainchainTokenID = getMainchainTokenID(ctx.chainID);
+
+		const { channelData } = chainInfo;
+
+		// channelData.messageFeeTokenID == Token.getTokenIDLSK();
+		if (!channelData.messageFeeTokenID.equals(mainchainTokenID)) {
+			throw new Error(`channelData.messageFeeTokenID is not equal to Token.getTokenIDLSK().`);
+		}
+
+		// channelData.minReturnFeePerByte == MIN_RETURN_FEE_PER_BYTE_LSK.
+		if (channelData.minReturnFeePerByte !== MIN_RETURN_FEE_PER_BYTE_BEDDOWS) {
+			throw new Error(
+				`channelData.minReturnFeePerByte is not equal to MIN_RETURN_FEE_PER_BYTE_BEDDOWS.`,
+			);
+		}
+	}
+
+	private _verifyChainValidators(chainInfo: ChainInfo) {
+		const { chainValidators, chainData } = chainInfo;
+		const { activeValidators, certificateThreshold } = chainValidators;
+
+		// activeValidators must have at least 1 element and at most MAX_NUM_VALIDATORS elements
+		if (activeValidators.length === 0 || activeValidators.length > MAX_NUM_VALIDATORS) {
```

Comment:
I think in the implementation you added this into the schema, (file schemas.ts, line 109) so perhaps this check can be avoided here (unless you want to throw a specific error for this -- but then it could be removed from schema?)

Reply:
I see now you removed this check, but also commented out the constraint on `MAX_NUM_VALIDATORS` (in shcemas.ts file), so now it is not checked at all..

Reply:
Commented here: https://github.com/LiskHQ/lisk-sdk/pull/8238/files#r1142049348